src/com/facebook/buck/jvm/kotlin/KotlinBuckConfig.java

```diff
@@ -20,26 +20,117 @@
 import com.facebook.buck.io.ExecutableFinder;
 import com.facebook.buck.rules.HashedFileTool;
 import com.facebook.buck.rules.Tool;
+import com.facebook.buck.util.HumanReadableException;
 import com.google.common.base.Supplier;
 import com.google.common.base.Suppliers;
 
+import java.io.IOException;
+import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.Optional;
 
 public class KotlinBuckConfig {
+  private static final String SECTION = "kotlin";
+
   private static final Path DEFAULT_KOTLIN_COMPILER = Paths.get("kotlinc");
 
   private final BuckConfig delegate;
+  private Path kotlinHome;
 
   public KotlinBuckConfig(BuckConfig delegate) {
     this.delegate = delegate;
   }
 
+  /**
+   * Get the Tool instance for the Kotlin compiler.
+   * @return the Kotlin compiler Tool
+   */
   public Supplier<Tool> getKotlinCompiler() {
-    Path compilerPath = delegate.getPath("kotlin", "compiler").orElse(DEFAULT_KOTLIN_COMPILER);
+    Path compilerPath = getKotlinHome().resolve("kotlinc");
+    if (!Files.isExecutable(compilerPath)) {
+      compilerPath = getKotlinHome().resolve(Paths.get("bin", "kotlinc"));
+      if (!Files.isExecutable(compilerPath)) {
+        throw new HumanReadableException("Could not resolve kotlinc location.");
+      }
+    }
 
     Path compiler = new ExecutableFinder().getExecutable(compilerPath, delegate.getEnvironment());
-
     return Suppliers.ofInstance(new HashedFileTool(compiler));
   }
+
+  /**
+   * Get the path to the Kotlin runtime jar.
+   * @return the Kotlin runtime jar path
+   */
+  public Path getPathToRuntimeJar() {
+    Optional<String> value = delegate.getValue(SECTION, "runtime_jar");
+    boolean isAbsolute = (value.isPresent() && Paths.get(value.get()).isAbsolute());
+
+    Path runtime = delegate.getPath(SECTION, "runtime_jar", !isAbsolute).orElse(null);
+    if (runtime != null) {
+      return runtime.normalize();
+    }
+
+    runtime = getKotlinHome().resolve("kotlin-runtime.jar");
+    if (Files.isRegularFile(runtime)) {
+      return runtime.normalize();
+    }
+
+    runtime = getKotlinHome().resolve(Paths.get("lib", "kotlin-runtime.jar"));
+    if (Files.isRegularFile(runtime)) {
+      return runtime.normalize();
+    }
+
+    throw new HumanReadableException("Could not resolve kotlin runtime JAR location.");
+  }
+
+  private synchronized Path getKotlinHome() {
```

Comment:
Probably coding too late at night ... because of the check below my brain switched into "what if two threads are accessing this instance" mode.  But actually that does not seem possible.  If that is true, and you know better than I, then I can take the synchronized out.